x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/AggregateExec.java

```diff
@@ -7,57 +7,60 @@
 
 package org.elasticsearch.xpack.esql.plan.physical;
 
+import org.elasticsearch.compute.aggregation.AggregatorMode;
 import org.elasticsearch.xpack.esql.core.expression.Attribute;
+import org.elasticsearch.xpack.esql.core.expression.AttributeSet;
 import org.elasticsearch.xpack.esql.core.expression.Expression;
-import org.elasticsearch.xpack.esql.core.expression.Expressions;
 import org.elasticsearch.xpack.esql.core.expression.NamedExpression;
 import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
 import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.plan.logical.Aggregate;
 
 import java.util.List;
 import java.util.Objects;
 
 public class AggregateExec extends UnaryExec implements EstimatesRowSize {
     private final List<? extends Expression> groupings;
     private final List<? extends NamedExpression> aggregates;
+    /**
+     * The output attributes of {@link AggregatorMode#INITIAL} aggregations, resp.
+     * the input attributes of {@link AggregatorMode#FINAL} aggregations.
+     */
+    private final List<Attribute> intermediateAttributes;
```

Comment:
It looks like `AggregateExec` is only ever serialized when we use data node level reduction, which is currently barred behind a pragma and so cannot be used, so this change should be fine.

We may have to keep track on a more granular level, though: for physical optimizations to correctly respect attribute dependencies, we need to know which aggregates/groupings are responsible for which intermediate attributes.

**Note to self: double check serialization one more time before merging.**